perun-dispatcher/src/main/java/cz/metacentrum/perun/dispatcher/scheduling/PropagationMaintainer.java

```diff
@@ -82,8 +91,62 @@ public void setDispatcherProperties(Properties dispatcherProperties) {
 		}
 	}
 
+	public ThreadPoolTaskScheduler getPerunScheduler() {
+		return perunScheduler;
+	}
+	
+	public void setPerunScheduler(ThreadPoolTaskScheduler perunScheduler) {
+		this.perunScheduler = perunScheduler;
+	}
+
 	// ----- methods -------------------------------------
 
+	private class TaskExecution implements Runnable {
+		private Task task;
+	
+		public TaskExecution(Task task) {
+			super();
+			this.task = task;
+		}
+		
+		@Override
+		public void run() {
+			if(task.getStatus().equals(TaskStatus.DONE)) {
```

Comment:
You should probably schedule all Tasks in other ending states too: ``TaskStatus.ERROR, TaskStatus.GENERROR, TaskStatus.SENDERROR``. Other states are considered "processing".

Reply:
I disagree with that. Tasks that ended in one of the error states should be rescheduled according to the default rules for failed tasks - ie. with task delay and recurrence. Only after succesfull completion they should be subject to the periodic scheduling again.

Reply:
I'm not saying, that its wrong in the principle, but it means, that if Task fails, It will perform just 2 more tries in cca 10 minutes (our default) and then it will wait 2 days for the next try, since these Tasks shouldn't be triggered by source data changes, but the cron only. 
If there is some kind of temporary fail and cron is set to "every hour", it won't try to start the propagation for the 2 whole days and I would expect that "try" to be every hour, just like the cron is set. I'll think about it more and discuss it within the core team.

Reply:
Then what about scheduling error tasks by cron only after they have exhausted the retries?